### PR TITLE
fix(): added verification for package names

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

File: src/api/auth.ts [hunk 1]
```diff
@@ -118,6 +118,12 @@ export function getTeamUsers(team: string, organization: string, auth: any): any
   return users;
 }
 
+export function checkPackageName(pkgName: string): boolean {
+  let regexStart = /^[a-z0-9~()'!*:@,;+?-]{1}/g;
```
Comment (jkuri) [inline src/api/auth.ts]: this regexes cannot work, you are allowing all of `~()'!*:@,;+?-` characters in a package name. where did you get this? help yourself with http://regexr.com/.

Comment (jkuri) [reply src/api/auth.ts]: woops, sorry, I didn't saw `^` at the beggining.

Comment (Izak88) [reply src/api/auth.ts]: I know, npm documentation https://docs.npmjs.com/files/package.json says `the name can't contain any non-URL-safe characters`, which means that `$-_.+!*'(),` should be okay... Although you're right, it really shouldn't contain `~:?` I don't know of course we can be more strict than npm about package names, what do you think?

Comment (jkuri) [reply src/api/auth.ts]: yes, I think `~:?` shouldn't be allowed. can you maybe make single regex for this checking?

Comment (Izak88) [reply src/api/auth.ts]: Why do you think that would be better? I'll remove `~:?` from each regex and check the whole word. If there is a match that means that this characters doesn't exists there. I'll add example like that in unit test...

**Comment by jkuri** [comment]
the main build is failing. please check why.

**Comment by Izak88** [comment]
the problem is in `Promise.reject()` it should return something in reject, I'll fix that...

**Comment by Izak88** [comment]
Fixed

**Review by jkuri — APPROVED** [review]

**Pull request merged**